src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private int arraySize = DEFAULT_CAPACITY;
+    private T[] dataValue;
+    private int size;
+
+    public ArrayList() {
+        dataValue = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int newCapacity) {
+        dataValue = (T[]) new Object[newCapacity];
+    }
+
     @Override
     public void add(T value) {
-
+        resize();
+        dataValue[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 & index <= size) {
+            resize();
+            System.arraycopy(dataValue, index, dataValue, index + 1, size - index);
+            dataValue[index] = value;
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index isn't valid.");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            resize();
+            dataValue[size] = list.get(i);
+            size++;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+
+        if (index < size && index >= 0) {
+            return dataValue[index];
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't exist.");
+        }
```

Comment:
```suggestion
        if (index >= size && index < 0) {
        	throw new ArrayListIndexOutOfBoundsException("Index " + index + " doesn't exist sor size " + size);
        }
            return dataValue[index];
```

it's better to make exception messages more informative